core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -331,6 +333,9 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
             if (get.realtime()) {
+                if (isRealtimeGetEnabled == false) {
+                    throw new IllegalArgumentException("get with realtime=true is disabled on this index");
```

Comment:
since realtime=true is the default, maybe the message should rather say something like "realtime gets are disabled. It is still possible to perform non-realtime gets by passing realtime=false"?